qemu/tests/libvirt_host_model_test.py

```diff
@@ -0,0 +1,175 @@
+        f.write(xml)
+    try:
+        logging.info("Get cpu model and features from virsh")
+        virsh.define(xml_file)
+        virsh.start(name)
+    except Exception as err:
+        test.cancel(err)
+    else:
+        qemu_cpu_info = cpu.get_cpu_info_from_virsh_qemu_cli(name)
+        libvirt_cpu_info = get_cpu_info_from_dumpxml(name)
+        params["qemu_cpu_info"] = qemu_cpu_info
+        params["libvirt_cpu_info"] = libvirt_cpu_info
+
+        cpu_model_qemu = qemu_cpu_info["model"]
+        cpu_model_libvirt = libvirt_cpu_info["model"]
+        params["qemu_proc_cpu_flags"] = qemu_cpu_info["flags"]
+        if cpu_model_qemu != cpu_model_libvirt:
+            logging.error("mismatch cpu model bwteen qemu %s and libvirt %s",
+                          cpu_model_qemu, cpu_model_libvirt)
+            return False
+        params["cpu_model"] = cpu_model_qemu
+        qemu_cpu_flags = cpu.parse_qemu_cpu_flags(qemu_cpu_info['flags'])
+        libvirt_cpu_flags = libvirt_cpu_info['features']
+        qemu_cpu_flags = extend_flags_patterns(qemu_cpu_flags)
+        exclude_map = eval(params.get('exclude_map', '{}'))
+        check_exclude = False
+        if cpu_model_qemu in exclude_map.keys():
+            exclude_map_flags = exclude_map[cpu_model_qemu]
+            check_exclude = True
+        miss_flags = []
+        mismatch_flags = []
+        result_bool = True
+        for flag in libvirt_cpu_flags.keys():
+            if flag not in qemu_cpu_flags.keys():
+                if libvirt_cpu_flags[flag] == 'on':
+                    miss_flags.append(flag)
+            elif libvirt_cpu_flags[flag] != qemu_cpu_flags[flag]:
+                mismatch_flags.append(flag)
+        if miss_flags:
+            logging.error("\nmiss flags %s from qemu cli\n", miss_flags)
+            for miss_flag in miss_flags:
+                if check_exclude and miss_flag not in exclude_map_flags:
+                    result_bool = False
+                    break
+        if mismatch_flags:
+            logging.error("\nmismatch flags %s between libvirt and qemu\n",
+                          mismatch_flags)
+            for mismatch_flag in miss_flags:
+                if check_exclude and mismatch_flag not in exclude_map_flags:
+                    result_bool = False
+                    break
+        return result_bool
+    finally:
+        if virsh.is_alive(name):
+            virsh.destroy(name, ignore_status=True)
+        virsh.undefine(name, ignore_status=True)
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test cpu models with full flags that libvirt generates
+    1) start a default configuration xml by virsh
+    2) get the cpu elements from libvirt xml by 'virsh dumpxml $domain'
+    3) parse the cpu_flags and store into libvirt_cpu_flags
+    4) get the cpu flags from qemu process and stroe to qemu_cpu_flags
+    5) Compare compare the cpu flags, model of xml and qemu process, should be same
```

Comment:
Redundant word 'compare'.

Reply:
And in step 6, it's actually combined flags.

Reply:
> And in step 6, it's actually combined flags.

Thanks and update.